### PR TITLE
gdscript-comint.el: add `gdscript-godot-executable' support flatpak 

### LINKED ISSUE #119: "open-project-in-editor" Flatpak Support
I don't know if this is possible, due to the sand-boxed nature of flatpaks, but I would like to be able to use the `M-x gdscript-godot-open-project-in-editor` to launch the flatpak version of Godot Engine.

I tried a workaround by creating a shell script that launches the flatpak and setting `(setq gdscript-godot-executable "path/to/godot")` to the script, but it does not works right. When I try opening or running the project through emacs, it just opens the Project Manager

This is the shell script, it just launches the Godot engine flatpak
```sh
#!/usr/bin/env bash

flatpak run org.godotengine.Godot
```

### PR DESCRIPTION
**Pull request by xiliuya**
Fix #119 .

### CODE REVIEW

File: gdscript-comint.el [hunk 1]
```diff
@@ -58,8 +58,17 @@
...
-        (inhibit-read-only t))
-    (if (not (or (file-executable-p gdscript-godot-executable) (executable-find gdscript-godot-executable)))
+        (inhibit-read-only t)
+        (godot-command gdscript-godot-executable))
+
+    ;; support flatpak
+    (if (string-match-p (rx "run" (* space) "org.godotengine.Godot") godot-command)
+        (let ((exec-split (split-string godot-command)))
+          (setq godot-command (car exec-split))
+          (setq arguments (append (cdr exec-split) arguments ))))
```
Comment (jcs090218) [inline gdscript-comint.el:68]: We can use `when` here.

Make sure to remove trailing space after `arguments`.

**Review by xiliuya — COMMENTED** [review]
add `gdscript-godot-executable' support flatpak .

**Comment by xiliuya** [comment]
@jcs090218 Can you help  approve this? Thanks! :smile:

**Review by jcs090218 — APPROVED** [review]
LGTM. ;)

**Pull request merged**